src/KeyringController.ts

```diff
@@ -317,8 +317,10 @@ export class KeyringController extends BaseController<BaseConfig, KeyringState>
 	async fullUpdate() {
 		const keyrings = await Promise.all(
 			this.keyring.keyrings.map(async (keyring: KeyringObject, index: number) => {
+				const keyringAccounts = await keyring.getAccounts();
+				const accounts = keyringAccounts.length ? keyringAccounts.map((address) => toChecksumAddress(address)) : [];
```

Comment:
I think this check should use `Array.isArray` instead of checking `length` (which works by accident despite returning `false` for an array with no items.)